diff --git a/entity_print.module b/entity_print.module
index f57ceca..c6acf94 100644
--- a/entity_print.module
+++ b/entity_print.module
@@ -5,6 +5,8 @@
  * Print any entity.
  */
 
+require_once('lib/dompdf-dompdf/v0.8.3/autoload.inc.php');
+
 use Drupal\Core\Entity\ContentEntityInterface;
 use Drupal\Core\Entity\Display\EntityViewDisplayInterface;
 use Drupal\Core\Entity\EntityInterface;
diff --git a/ludwig.json b/ludwig.json
index e69de29..3f0bc5e 100644
--- a/ludwig.json
+++ b/ludwig.json
@@ -0,0 +1,8 @@
+{
+  "require": {
+    "dompdf/dompdf": {
+      "version" : "v0.8.3",
+      "url": "https://github.com/dompdf/dompdf/releases/download/v0.8.3/dompdf_0-8-3.zip"
+    }
+  }
+}
\ No newline at end of file
